server/src/server.js

```diff
@@ -645,6 +645,14 @@ app.get("/images/:imageid", function (req, res) {
         anonAnalytics.pageview(analyticsUrl).send();
       }
       res.header("Content-Type", obj.contentType);
+      if (req.query.filename) {
+        let extension = "." + obj.contentType.split("/").pop();
+        let filename= req.query.filename;
+        if (! filename.endsWith(extension)) {
+          filename = filename + extension;
+        }
+        res.header("Content-Disposition", `attachment; filename="${filename}"`)
+      }
```

Comment:
This allows a potential attacker to construct a link that makes someone download the image with whatever filename they wish (though with the necessary extension).  We shouldn't allow that.

I don't think we need any server support for this, the [download](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a) attribute is sufficient for what we're doing here.